R/model-tree.R

```diff
@@ -681,6 +733,37 @@ color_tree_by <- function(tree_data, color_by = "run"){
   return(tree_data)
 }
 
+#' Create a size column based on the unique _numeric_ (or integer) values of
+#' another column.
+#' @inheritParams make_tree_tooltip
+#' @inheritParams model_tree
+#' @noRd
+size_tree_by <- function(tree_data, size_by = NULL){
+  if(!is.null(size_by)){
+    checkmate::assert_true(size_by %in% names(tree_data))
+    tree_data$node_size <- tree_data[[size_by]]
+    # Scale size with numeric value
+    if(inherits(tree_data$node_size, c("numeric", "integer"))){
+      # Set node sizes with NA values (including start node) to mean value
+      mean_val <- mean(tree_data$node_size, na.rm = TRUE)
+      tree_data$node_size[is.na(tree_data$node_size)] <- mean_val
+    }else{
+      col_class <- class(tree_data[[size_by]])
+      cli::cli_warn(
+        c(
+          "Only numeric columns are supported. Column {.val {size_by}} is {.val {col_class}}",
+          "i" = "Setting node size to constant"
+        )
+      )
+      tree_data$node_size <- 1
+    }
+  }else{
+    tree_data$node_size <- 1
+  }
```

Comment:
Think it's fine to leave, but what's your thinking behind setting it to a constant size rather than warning and leaving the default "leafCount" behavior?

Reply:
(Sorry, I put in the wrong comment; I've replaced the text.)

Reply:
Addressed in [e2c0909](https://github.com/metrumresearchgroup/bbr/pull/720/commits/e2c0909c88736176ccee06a70505f66231115e22). Will mark this one as resolved for now, and any follow up regarding either this or the previous comment can be discussed in that thread since the referenced commit refactored both sections in the same way.